documentation/gnmi/dash-gnmi-design.md

```diff
@@ -0,0 +1,144 @@
+# GNMI Interface for DASH
+
+## Overview
+
+SONiC should provide a gNMI server interface for DASH project.
+Below diagram shows the architecture of the DASH project, the host server and DPU card are connected through PCIE, and together they form a smart switch.
+
+![dash-arch](./images/gnmi-arch.svg)
+
+Both host server and DPU card are running SONiC image. GNMI server and redis for APPL_DB are on host server, orchagent is on DPU card, GNMI server will use ZMQ over TCP to communicate with orchagent and redis.
+
+### Requirements
+
+* Set and get RPCs must be supported. Customers will use get RPC to retrieve DASH configurations and use set PRC to apply new DASH configurations.
+* Need to configure huge DASH table entries to APPL_DB, with high speed.
+* Minimal redis table scaling requirements:
+
+|Item |Expected value |
+|--|--|
+|VNETs|1024|
+|ENI|64 per card|
+|Outbound Routes per ENI|100K|
+|Inbound Routes per ENI|10K|
+|NSGs per ENI|10|
+|ACL rules per NSG|1000|
+|ACL prefixed per ENI|10*100K|
+|Max prefixed per rule|8K|
+|ACL ports per ENI|10*10K SRC/DST ports|
+|CA-PA mapping|10M per card|
```

Comment:
I feel it's better to link here: https://github.com/sonic-net/DASH/blob/main/documentation/general/dash-sonic-hld.md#14-scaling-requirements . Because the scaling table is continuously updated.

Reply:
updated.